src/Runners/PHPUnit/BaseRunner.php

```diff
@@ -95,11 +99,37 @@ private function load(SuiteLoader $loader): void
         $loader->load();
         $executables   = $this->options->functional() ? $loader->getTestMethods() : $loader->getSuites();
         $this->pending = array_merge($this->pending, $executables);
+        $this->sortPending();
+
         foreach ($this->pending as $pending) {
             $this->printer->addTest($pending);
         }
     }
 
+    private function sortPending(): void
+    {
+        $seed = $this->options->randomOrderSeed();
+        if ($seed > 0 && $this->options->orderBy() === Options::RANDOM_ORDER) {
+            mt_srand($seed);
+
+            $keys      = array_keys($this->pending);
+            $new_array = [];
+            shuffle($keys);
```

Comment:
I see that `shuffle` doesn't retain keys.
I'm going to try removing keys in `$this->pending` so we can call `shuffle($this->pending)` directly.

Reply:
https://github.com/paratestphp/paratest/pull/542 merged, you can now rebase and ease this part

Reply:
Rebased